src/org/labkey/test/tests/DomainDesignerTest.java

```diff
@@ -835,6 +835,58 @@ public void setFieldAsVariable() throws Exception
         assertEquals("extraField should not have recommendedVariable marked true", false, testColumnProperties.get("extraField"));
     }
 
+    @Test
+    public void testLookUpFieldSameContainer() throws IOException, CommandException
+    {
+        String sampleSet = "setFieldAsLookup";
+        String listName = "lookUpList";
+
+        FieldDefinition.LookupInfo lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "lists", listName);
+        TestDataGenerator dgen1 = new TestDataGenerator(lookupInfo)
+                .withColumnSet(List.of(
+                        TestDataGenerator.simpleFieldDef("name", FieldDefinition.ColumnType.String),
+                        TestDataGenerator.simpleFieldDef("color", FieldDefinition.ColumnType.String)));
+        DomainResponse createResponse = dgen1.createDomain(createDefaultConnection(true), "IntList", Map.of("keyName", "id"));
+
+        lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "exp.materials", sampleSet);
+        TestDataGenerator dgen = new TestDataGenerator(lookupInfo)
+                .withColumnSet(List.of(
+                        TestDataGenerator.simpleFieldDef("name", FieldDefinition.ColumnType.String),
+                        TestDataGenerator.simpleFieldDef("extraField", FieldDefinition.ColumnType.String),
+                        TestDataGenerator.simpleFieldDef("testCol", FieldDefinition.ColumnType.String)));
+        createResponse = dgen.createDomain(createDefaultConnection(true), "SampleSet");
+
+        DomainDesignerPage domainDesignerPage = DomainDesignerPage.beginAt(this, getProjectName(), "exp.materials", sampleSet);
+        DomainFormPanel domainFormPanel = domainDesignerPage.fieldProperties(sampleSet);
+
+        DomainFieldRow lookUpRow = domainFormPanel.addField("lookUpField")
+                .setType("Lookup")
+                .expand()
+                .setFromFolder("Current Folder")
+                .setFromSchema("lists")
+                .setFromTargetTable("lookUpList (Integer)")
+                .setDescription("LookUp in same container")
+                .clickCancelCross();
+
+        assertEquals("Incorrect detail message","/DomainDesignerTest Project > lists > lookUpList",lookUpRow.detailsMessage());
+
+        domainDesignerPage.saveButton().click();
```

Comment:
Where possible, use domainDesignerPage.clickSave (because it also handles the waiting for following stuff like errors or page navigation) rather than accessing the button and clicking it directly- that way you can reduce the risk of painful race conditions